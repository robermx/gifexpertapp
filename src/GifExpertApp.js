import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h1 className="center text-light"> GifExpertApp </h1>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ul>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ul>
    </>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default GifExpertApp;
