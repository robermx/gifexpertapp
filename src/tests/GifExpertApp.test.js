// import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas de GifExpertApp componente principal', () => {
  test('Debe mostrarse adecuadamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de mostar una lista de categorÃ­as', () => {
    const categories = ['bachas', 'tachas'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
