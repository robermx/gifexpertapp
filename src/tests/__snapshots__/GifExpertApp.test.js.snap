// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifExpertApp componente principal Debe de mostar una lista de categor√≠as 1`] = `
<Fragment>
  <h1
    className="center text-light"
  >
     GifExpertApp 
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="bachas"
      key="bachas"
    />
    <GifGrid
      category="tachas"
      key="tachas"
    />
  </ul>
</Fragment>
`;

exports[`Pruebas de GifExpertApp componente principal Debe mostrarse adecuadamente 1`] = `
<Fragment>
  <h1
    className="center text-light"
  >
     GifExpertApp 
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
